sparse/core.py

```diff
@@ -603,55 +613,238 @@ def elemwise_binary(self, func, other, *args, **kwargs):
+                                                                                          other_params,
+                                                                                          result_shape)
+
+            other_broadcast_coords, other_broadcast_data = self._get_broadcast_coords_data(other_coords[matched_other],
+                                                                                           matched_coords,
+                                                                                           other_data[matched_other],
+                                                                                           other_params,
+                                                                                           result_shape)
+
+            data_list.append(func(0, other_unmatched_data,
+                                  *args, **kwargs))
+            coords_list.append(other_unmatched_coords)
+            data_list.append(func(0, other_broadcast_data,
+                                  *args, **kwargs))
+            coords_list.append(other_broadcast_coords)
 
         # Concatenate matches and mismatches
-        data = np.concatenate([func(self_data[matched_self],
-                                    other_data[matched_other],
-                                    *args, **kwargs),
-                               func(self_data[unmatched_self], 0,
-                                    *args, **kwargs),
-                               func(0, other_data[unmatched_other],
-                                    *args, **kwargs)])
-        coords = np.concatenate([self_coords[matched_self],
-                                 self_coords[unmatched_self],
-                                 other_coords[unmatched_other]])
+        data = np.concatenate(data_list) if len(data_list) else np.empty((0,), dtype=self.dtype)
+        coords = np.concatenate(coords_list, axis=1) if len(coords_list) else np.empty((0, len(result_shape)),
+                                                                                       dtype=self.coords.dtype)
 
         nonzero = data != 0
         data = data[nonzero]
-        coords = coords[nonzero]
+        coords = coords[:, nonzero]
+
+        return COO(coords, data, shape=result_shape, has_duplicates=False)
+
+    @staticmethod
+    def _get_broadcast_shape(shape1, shape2, is_result=False):
+        """
+        Get the overall broadcasted shape from two different shapes, without actually constructing any arrays.
+        """
+        # https://stackoverflow.com/a/47244284/774273
+        if not all((l1 == l2) or (l1 == 1) or ((l2 == 1) and not is_result) for l1, l2 in zip(shape1[::-1],
+                                                                                             shape2[::-1])):
+            raise ValueError('operands could not be broadcast together with shapes %s, %s' % (shape1, shape2))
```

Comment:
Can I request that we wrap code shorter than this when possible?  We allow lines up to 120 in flake8, but in practice we try to aim shorter.  120 is just as lenient as we're willing to be, not our target.

Reply:
Sure, I'll try to cut it down to 80.